src/utils.js

```diff
@@ -1,241 +1,227 @@
-		return string.toLowerCase();
-	} else if (type === "sentence") {
-		string += "";
-		first = string.charAt(0).toUpperCase();
-		return first + string.substring(1);
-	} else if (type === "formal") {
-		words = string.split(/\s+/);
-		result = [];
-		for (i = 0, len = words.length; i < len; i++) {
-			word = words[i];
-			first = word.charAt(0).toUpperCase();
-			result.push(first + word.substring(1));
-		}
-		return result.join(" ");
-	}
-	return content;
-};
-
-/**
-[]string parseCallArgs
-
-Parse a string and return shell-like arguments as an array. Normally this
-means each word in the string becomes an item in the result, but quoted
-sections of the input will come back as a single item.
-
-Example:
-
-```javascript
-console.log( parseCallArgs('please google "writing chat bot"'));
-// ["please", "google", "writing chat bot"]
-```
-*/
-exports.parseCallArgs = function(str) {
-	var buff, c, doubleQuoteRe, flushBuffer, i, insideAString, len, result, spaceRe;
-	result = [];
-	buff = "";
-	insideAString = false;
-	spaceRe = /\s/ig;
-	doubleQuoteRe = /"/ig;
-	flushBuffer = function() {
-		if (buff.length !== 0) {
-			result.push(buff);
-		}
-		return buff = "";
-	};
-	for (i = 0, len = str.length; i < len; i++) {
-		c = str[i];
-		if (c.match(spaceRe) && !insideAString) {
-			flushBuffer();
-			continue;
-		}
-		if (c.match(doubleQuoteRe)) {
-			if (insideAString) {
-				flushBuffer();
-			}
-			insideAString = !insideAString;
-			continue;
-		}
-		buff += c;
-	}
-	flushBuffer();
-	return result;
-};
-
-/**
-object clone (object)
-
-Clone an object.
-*/
-exports.clone = function(obj) {
-	var copy, key;
-	if (obj === null || typeof obj !== "object") {
-		return obj;
-	}
-	copy = obj.constructor();
-	for (key in obj) {
-		if (!obj.hasOwnProperty(key)) {
-			continue;
-		}
-		copy[key] = exports.clone(obj[key]);
-	}
-	return copy;
-};
-
-/**
-int nIndexOf (string, string match, int index)
-
-Finds a match in a string at a given index
-
-Usage:
-string = "My name is Rive"
-match = " "
-index = 2
-return = 7
-
-Summary: It will look for a second space in the string
-*/
-exports.nIndexOf = function(string, match, index) {
-	return string.split(match, index).join(match).length;
-};
+ * Object clone (object)
+ *
+ * Clone an object.
+ */
+
+const clone = (obj) => {
+    if (obj === null || typeof obj !== "object") {
+      return obj
+    }
+    const copy = obj.constructor()
+    for (const key in obj) {
+      if (!obj.hasOwnProperty(key)) {
+        copy[key] = clone(obj[key])
+      }
+    }
+    return copy
+  },
+
+  /**
+   * Void extend (object a, object b)
+   *
+   * Combine the properties of both objects into one. The properties from
+   * object 'b' are inserted into 'a'.
+   */
+
+  extend = (a, b) => (() => {
+    const result = []
+    for (const attr in b) {
+      const value = b[attr]
+      if (!b.hasOwnProperty.call(attr)) {
+        result.push(a[attr] = value)
+      }
+    }
+    return result
+  })(),
```

Comment:
Why is this an IIFE here? Seems unnecessary. Is it some kind of odd optimization?

Reply:
This is decaffeination glitch, didn't saw it. Looks like unnecessary and totally removable.